src/py42/clients/trustedactivities.py

```diff
@@ -0,0 +1,83 @@
+from py42.choices import Choices
+
+
+class TrustedActivityType(Choices):
+    DOMAIN = "DOMAIN"
+    SLACK = "SLACK"
+
+
+class TrustedActivitiesClient:
+    """A client to expose the trusted activities/data preferences API
+
+    `Rest documentation <https://developer.code42.com/api/#tag/Trusted-Activities>`__
+    """
+
+    def __init__(self, trusted_activities_service):
+        self._trusted_activities_service = trusted_activities_service
+
+    def get_all(self, type=None):
+        """Gets all trusted activities.
+        `Rest documentation <https://developer.code42.com/>`
+
+        Args:
+            type (str, optional): Type of the trusted activity. `TrustedActivityType.DOMAIN` or `TrustedActivityType.SLACK`
+
+        Returns:
+            :class:'py42.response.Py42Response'
+        """
+        return self._trusted_activities_service.get_all(type)
+
+    def create(self, type, value, description=None):
+        """Gets all trusted activities with the given type.
+        `Rest documentation <https://developer.code42.com/>`
+
+        Args:
+            type (str): Type of the trusted activity. `TrustedActivityType.DOMAIN` or `TrustedActivityType.SLACK`
+            value (str): The URL of the domain or name of the Slack workspace.
+            description (str, optional): Description of the trusted activity.
+
+        Returns:
+            :class:'py42.response.Py42Response'
+        """
+        return self._trusted_activities_service.create(type, value, description)
+
+    def get(self, id):
+        """Retrieve trusted activity details by given resource number.
+        `Rest documentation <https://developer.code42.com/>`
+
+        Args:
+            id (int): Resource number of the trusted activity or domain.
+
+        Returns:
+            :class:'py42.response.Py42Response'
+        """
+        return self._trusted_activities_service.get(id)
+
+    def update(self, id, type=None, value=None, description=None):
```

Comment:
Users can't change the `type` of a trusted activity (currently throws a custom `invalid change` exception), so should this not be a parameter of the `update()` method in the client, or altogether (remove it from `update() in both the client and service)?

Reply:
yeah, I would remove it, no reason to provide functionality that will either do nothing or always fail